src/conn/conn_dhandle.c

```diff
@@ -17,7 +17,11 @@ __conn_dhandle_destroy(WT_SESSION_IMPL *session, WT_DATA_HANDLE *dhandle)
 {
 	WT_DECL_RET;
 
-	WT_WITH_DHANDLE(session, dhandle, ret = __wt_btree_discard(session));
+	if (WT_PREFIX_MATCH(dhandle->name, "file:"))
```

Comment:
That would be fine with me. It feels like an implementation showing itself in the interface, but I'd rather that was explicit than peeking under the covers like the string comparison is doing.